lib/view/community/detailed_meetup_view.dart

```diff
@@ -59,301 +46,390 @@ class _DetailedMeetupViewState extends State<DetailedMeetupView> {
-  Future<bool> _handleTapRSVP(bool isLiked) async {
-    setState(() {
-      if (isLiked) {
-        widget.meetup.removeRsvpAttendee(widget.currUser);
-      } else {
-        widget.meetup.addRsvpAttendee(widget.currUser);
-      }
-    });
-    return !isLiked;
-  }
-
   @override
   Widget build(BuildContext context) {
-    return Center(
-      child: Card(
-        child: Column(mainAxisSize: MainAxisSize.min, children: <Widget>[
-          ListTile(
-            title: Text(widget.meetup.title ?? "Untitled Meetup",
-                style: DefaultTextStyle.of(context)
-                    .style
-                    .apply(fontSizeFactor: 1.8, fontWeightDelta: 2)),
-            trailing: (widget.currUser.id == widget.meetup.creator.id &&
-                    DateTime.now().isBefore(widget.meetup.timeOfMeetUp
-                        .subtract(Duration(hours: 3))))
-                ? PopupMenuButton(
-                    onSelected: (value) {
-                      setState(() {
-                        if (value == "edit") {
-                          // TODO: navigate to edit meetup screen
-                        } else if (value == "delete") {
-                          widget.meetup
-                              .cancel(); // TODO: use manager function to delete meetup
-                        }
-                      });
-                    },
-                    itemBuilder: (context) => [
-                          PopupMenuItem(
-                            child: Row(
-                              children: const [
-                                Icon(Icons.edit, size: 20),
-                                SizedBox(width: 8),
-                                Text("Edit"),
-                              ],
+    return Consumer2<Meetup_Manager, User_Manager>(
+        builder: (context, meetupMgr, userMgr, child) {
+      return Center(
+        child: Card(
+          child: Column(mainAxisSize: MainAxisSize.min, children: <Widget>[
+            ListTile(
+              title: Text(
+                  meetupMgr.getMeetupByID(widget.meetupID).title ??
+                      "Untitled Meetup",
+                  style: DefaultTextStyle.of(context)
+                      .style
+                      .apply(fontSizeFactor: 1.8, fontWeightDelta: 2)),
+              trailing: (widget.currUserID ==
+                          meetupMgr.getMeetupByID(widget.meetupID).userID &&
+                      meetupMgr.canEditMeetup(widget.meetupID))
+                  ? PopupMenuButton(
+                      onSelected: (value) {
+                        setState(() {
+                          if (value == "edit") {
+                            // TODO ROUTING: navigate to edit meetup screen
+                          } else if (value == "delete") {
+                            meetupMgr.cancelMeetup(widget.meetupID);
+                            // TODO ROUTING: navigate to meetup list view
+                          }
```

Comment:
@RealDyllon could you also add the same routing from detailed thread view to detailed meetup view? thanks!

Reply:
done 👍🏼